gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GafferSpecBuilder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.gaas.factories;
+
+import uk.gov.gchq.gaffer.cache.impl.HashMapCacheService;
+import uk.gov.gchq.gaffer.cache.util.CacheProperties;
+import uk.gov.gchq.gaffer.controller.model.v1.GafferSpec;
+import uk.gov.gchq.gaffer.federatedstore.FederatedStore;
+import uk.gov.gchq.gaffer.federatedstore.FederatedStoreProperties;
+import uk.gov.gchq.gaffer.gaas.model.StoreType;
+import uk.gov.gchq.gaffer.proxystore.ProxyProperties;
+import uk.gov.gchq.gaffer.proxystore.ProxyStore;
+import uk.gov.gchq.gaffer.sketches.serialisation.json.SketchesJsonModules;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GafferSpecBuilder {
+
+    private String graphId;
+    private String description;
+    private Map<String, Object> storeProperties;
+    private boolean accumuloIsEnabled;
+
+    public GafferSpecBuilder graphId(final String graphId) {
+        this.graphId = graphId;
+        return this;
+    }
+
+    public GafferSpecBuilder description(final String description) {
+        this.description = description;
+        return this;
+    }
+
+    public GafferSpecBuilder storeProperties(final StoreType storeType) {
+        return storeProperties(storeType, null, null);
+    }
+
+    public GafferSpecBuilder storeProperties(final StoreType storeType, final String host, final String contextRoot) {
+        switch (storeType) {
+            case ACCUMULO:
+                // No AccumuloStoreProperties required for the graph
+                // Instead, enableAccumulo() in the GraphSpec to enable Accumulo store properties
+                throw new IllegalArgumentException("enableAccumulo() in the GraphSpec to enable Accumulo store properties");
```

Comment:
Rather than throwing an exception could you create an empty `stroreProperties` or leave it `null` and log a warning?
It's a bit messy having a `storeProperties` object and having a `enableAccumulo` method. It would be nice to have different objects for each store `GafferSpec`. That way you could get rid of the `enableAccumulo` and have the logic in an `AccumuloGraphStore` which extends `AbstractGraphSpec` (with the `graphId` and `description` logic in) and implements `GraphSpec`. You could then call the requested store type in the `GafferHelmValuesFactory.createGafferSpecFrom` method.

Reply:
- I'll remove it and use the abstract builder solution. I added the IAX just for future devs working on it if they add implementation in such a way they'd get the illegal arg feedback from code to hint how the helm Values is configured using Accumulo - use the abstract builder does and reads the same way more cleanly so will go with that

- I've created an `AbstractGafferSpecBuilder` for a 'generic' `GafferSpecBuilder` (used for all other stores) and `AccumuloGafferSpecBuilder` to extend, is this what you mean? 

I've pushed the code for this. I've left the default storeproperties values added in the builder as I made the assumption Users of the API will just want to create a Graph with a generic Proxy Store for example, and might not what StoreProps to use - would this be correct? 
In another ticket, I can add custom storeProperties to the request if another (more technical) User Group knows how store props work and would add their own custom properties for their graph